hivemind/optim/experimental/state_averager.py

```diff
@@ -631,7 +631,8 @@ def load_state_from_peers(self, **kwargs):
         Attempt to download the latest optimizer state from peers and update trainer parameters/statistics.
         :returns: whether or the averager succeeded in loading parameters
         """
-        main_parameters_and_extras = tuple(chain(self.main_parameters, self.extra_tensors))
+        opt_parameters = tuple(param for param_group in self.optimizer.param_groups for param in param_group["params"])
+        main_parameters_and_extras = tuple(chain(opt_parameters, self.extra_tensors))
```

Comment:
What does it mean? If `self.main_parameters` are invalid at this stage, shouldn't we replace them?